src/simulation.hpp

```diff
@@ -1517,8 +1737,14 @@ template <typename problem_t> void AMRSimulation<problem_t>::WritePlotFile() con
   amrex::Vector<amrex::MultiFab> mf = PlotFileMF();
   amrex::Vector<const amrex::MultiFab *> mf_ptr = amrex::GetVecOfConstPtrs(mf);
   amrex::Vector<std::string> varnames;
-  varnames.insert(varnames.end(), componentNames_cc_.begin(),
-                  componentNames_cc_.end());
+  varnames.insert(varnames.end(), componentNames_cc_.begin(), componentNames_cc_.end());
+  if constexpr (Physics_Indices<problem_t>::nvarTotal_fc > 0) {
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+      for (int icomp = 0; icomp < ncomp_fc_; ++icomp) {
```

Comment:
This code becomes simpler if you absorb the `AMREX_SPACEDIM` factor into `ncomp_fc_`.